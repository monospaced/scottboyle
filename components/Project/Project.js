import React, { Component } from "react";
import DocumentMeta from "react-document-meta";
import marked from "marked";
import Client from "../Client/Client";
import Visit from "../Visit/Visit";
import Screenshot from "../Screenshot/Screenshot";

class Project extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    const { data, route: { path } } = this.props;
    const project = data.projects[path];
    return (
      <main className="Main">
        <DocumentMeta
          title={`${data.title} | ${data.subtitle} | ${project.title}`}
        />
        <section>
          <h2>
            {project.title} <small>({project.date})</small>
          </h2>
          <div
            dangerouslySetInnerHTML={{ __html: marked(project.content.trim()) }}
          />
          {project.client && <Client client={project.client} />}
          {project.link && <Visit project={project} />}
          <Screenshot slug={path} project={project} />
        </section>
      </main>
    );
  }
}

module.exports = Project;
