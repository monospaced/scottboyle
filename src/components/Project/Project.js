import DocumentMeta from "react-document-meta";
import marked from "marked";
import React from "react";

import "../Main/Main.css";
import Client from "../Client/Client";
import Screenshot from "../Screenshot/Screenshot";
import Visit from "../Visit/Visit";

const Project = ({
  data: { description, projects, subtitle, title },
  route: { path },
}) => {
  const project = projects[path];
  const { client, content, date, link, title: projectTitle } = project;

  return (
    <main className="Main">
      <DocumentMeta
        title={`${title} | ${subtitle} | ${projectTitle}`}
        description={description}
      />
      <section>
        <h2>
          {projectTitle} <small>({date})</small>
        </h2>
        <div
          dangerouslySetInnerHTML={{ __html: marked.parse(content.trim()) }}
        />
        {client && <Client client={client} />}
        {link && <Visit project={project} />}
        <Screenshot project={project} slug={path} />
      </section>
    </main>
  );
};

export default Project;
