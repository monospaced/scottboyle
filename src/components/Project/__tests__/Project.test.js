import React from "react";
import { shallow } from "enzyme";

import data from "../../../scripts/__mocks__/data.js";
import Project from "../Project.js";

describe("Project component", () => {
  const { description, projects, subtitle, title } = data;

  it("should render correctly", () => {
    const props = {
      data: { description, projects, subtitle, title },
      route: {},
    };
    Object.keys(projects).map(key => {
      props.route.path = key;
      const component = shallow(<Project {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
