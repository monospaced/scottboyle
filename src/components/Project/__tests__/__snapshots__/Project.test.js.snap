// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component should render correctly 1`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | Thomson Reuters"
  />
  <section>
    <h2>
      Thomson Reuters
       
      <small>
        (
        2018
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Building a design system and React component&nbsp;library.</p>
",
        }
      }
    />
    <Client
      client={
        Object {
          "link": "https://monospaced.com",
          "title": "Monospaced",
        }
      }
    />
    <Visit
      project={
        Object {
          "client": Object {
            "link": "https://monospaced.com",
            "title": "Monospaced",
          },
          "content": "
Building a design system and React component&nbsp;library.
      ",
          "date": "2018",
          "image": Object {
            "height": 275,
            "width": 398,
          },
          "link": "https://www.thomsonreuters.com/en.html",
          "title": "Thomson Reuters",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": Object {
            "link": "https://monospaced.com",
            "title": "Monospaced",
          },
          "content": "
Building a design system and React component&nbsp;library.
      ",
          "date": "2018",
          "image": Object {
            "height": 275,
            "width": 398,
          },
          "link": "https://www.thomsonreuters.com/en.html",
          "title": "Thomson Reuters",
        }
      }
      slug="thomson-reuters"
    />
  </section>
</main>
`;

exports[`Project component should render correctly 2`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | Compass Group"
  />
  <section>
    <h2>
      Compass Group
       
      <small>
        (
        2017
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h3 id=\\"retail-pricing-portal\\">Retail Pricing Portal</h3>
<p>UI design and complete front-end build, Angular NgRx web&nbsp;app.</p>
",
        }
      }
    />
    <Client
      client={
        Object {
          "link": "https://monospaced.com",
          "title": "Monospaced",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": Object {
            "link": "https://monospaced.com",
            "title": "Monospaced",
          },
          "content": "
### Retail Pricing Portal

UI design and complete front-end build, Angular NgRx web&nbsp;app.
      ",
          "date": "2017",
          "image": Object {
            "height": 244,
            "width": 398,
          },
          "link": "",
          "title": "Compass Group",
        }
      }
      slug="compass-group"
    />
  </section>
</main>
`;

exports[`Project component should render correctly 3`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | Bilbary"
  />
  <section>
    <h2>
      Bilbary
       
      <small>
        (
        2012
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Prototype web e-book reader for digital library startup. Lives on as an open source project:&nbsp;<a href=\\"https://monospaced.github.io/paperback/\\">Paperback</a>.</p>
",
        }
      }
    />
    <Client
      client={
        Object {
          "link": "https://profitfromplay.com",
          "title": "Profit From Play",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": Object {
            "link": "https://profitfromplay.com",
            "title": "Profit From Play",
          },
          "content": "
Prototype web e-book reader for digital library startup. Lives on as an open source project:&nbsp;[Paperback](https://monospaced.github.io/paperback/).
      ",
          "date": "2012",
          "image": Object {
            "height": 190,
            "width": 199,
          },
          "link": "",
          "title": "Bilbary",
        }
      }
      slug="bilbary"
    />
  </section>
</main>
`;

exports[`Project component should render correctly 4`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | Genes Reunited"
  />
  <section>
    <h2>
      Genes Reunited
       
      <small>
        (
        2012
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Consulting and advising on a site-wide design refresh. Created indicative templates and assisted with&nbsp;integration.</p>
",
        }
      }
    />
    <Client
      client={
        Object {
          "link": "https://profitfromplay.com",
          "title": "Profit From Play",
        }
      }
    />
    <Visit
      project={
        Object {
          "client": Object {
            "link": "https://profitfromplay.com",
            "title": "Profit From Play",
          },
          "content": "
Consulting and advising on a site-wide design refresh. Created indicative templates and assisted with&nbsp;integration.
      ",
          "date": "2012",
          "image": Object {
            "height": 143,
            "width": 199,
          },
          "link": "http://genesreunited.co.uk",
          "title": "Genes Reunited",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": Object {
            "link": "https://profitfromplay.com",
            "title": "Profit From Play",
          },
          "content": "
Consulting and advising on a site-wide design refresh. Created indicative templates and assisted with&nbsp;integration.
      ",
          "date": "2012",
          "image": Object {
            "height": 143,
            "width": 199,
          },
          "link": "http://genesreunited.co.uk",
          "title": "Genes Reunited",
        }
      }
      slug="genes-reunited"
    />
  </section>
</main>
`;

exports[`Project component should render correctly 5`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | The National Lottery"
  />
  <section>
    <h2>
      The National Lottery
       
      <small>
        (
        2006
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Technical lead on an accessible Web Standards retrofit of key areas of national-lottery.co.uk, including the Lotto and EuroMillions purchase processes, user registration and account&nbsp;management.</p>
<p>Responsible for front-end architecture and all front-end development. Authored company Web Accessibility Guidelines, and conducted usability testing with screen-reader&nbsp;users.</p>
",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": null,
          "content": "
Technical lead on an accessible Web Standards retrofit of key areas of national-lottery.co.uk, including the Lotto and EuroMillions purchase processes, user registration and account&nbsp;management.

Responsible for front-end architecture and all front-end development. Authored company Web Accessibility Guidelines, and conducted usability testing with screen-reader&nbsp;users.
      ",
          "date": "2006",
          "image": Object {
            "height": 284,
            "width": 199,
          },
          "link": "",
          "title": "The National Lottery",
        }
      }
      slug="the-national-lottery"
    />
  </section>
</main>
`;

exports[`Project component should render correctly 6`] = `
<main
  className="Main"
>
  <SideEffect(DocumentMeta)
    description="Portfolio for Scott Boyle, freelance web developer based in London, UK."
    title="Scott Boyle | Portfolio | Age Concern"
  />
  <section>
    <h2>
      Age Concern
       
      <small>
        (
        2005
        )
      </small>
    </h2>
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h3 id=\\"age-concern-southwark\\">Age Concern Southwark</h3>
<p>Responsible for designing and developing an accessible <abbr title=\\"Disability Discrimination Act\\">DDA</abbr> compliant website for older&nbsp;users.</p>
",
        }
      }
    />
    <Client
      client={
        Object {
          "link": "",
          "title": "Two Plus Four",
        }
      }
    />
    <Screenshot
      project={
        Object {
          "client": Object {
            "link": "",
            "title": "Two Plus Four",
          },
          "content": "
### Age Concern Southwark

Responsible for designing and developing an accessible <abbr title=\\"Disability Discrimination Act\\">DDA</abbr> compliant website for older&nbsp;users.
      ",
          "date": "2005",
          "image": Object {
            "height": 305,
            "width": 199,
          },
          "link": "",
          "title": "Age Concern",
        }
      }
      slug="age-concern"
    />
  </section>
</main>
`;
